public/app/plugins/panel/datagrid/DataGridPanel.tsx

```diff
@@ -210,7 +210,14 @@ export function DataGridPanel({ options, data, id, fieldConfig, width, height }:
 
   const onRenameInputBlur = (columnName: string, columnIdx: number) => {
     const fields = frame.fields.map((f) => f);
-    fields[columnIdx].name = columnName;
+
+    if (fields[columnIdx].config.displayName) {
```

Comment:
rather than have this... does it make more sense to remove the configs?  (and perhaps try to remove any field overrides setting the name/display name?

I don't think things are going to land well if we try to reverse engineer the results that come out of:
```
      const name = getFieldDisplayName(field, frame, data);
```